src/builder/commands/copy.rs

```diff
@@ -186,27 +180,24 @@ impl BuildStep for Copy {
     }
 }
 
-fn reset_permissions(path: &Path, umask: u32)
-    -> Result<(), StepError>
+fn calc_mode(path: &Path, umask: u32)
```

Comment:
We already have metadata in the loop above. Just pass it here instead of a Path.